### PR TITLE
Switch to external manager for datastore operations

### PR DESCRIPTION
**Pull request by romeara**

### CODE REVIEW

File: corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java [hunk 1]
```diff
@@ -98,21 +92,21 @@ public Project create(ProjectRequest request) throws IOException {
...
+        return toProject(projectLocation);
     }
 
     @Override
     public Collection<Project> getAll() {
         Workspace workspace = workspaceService.getActiveWorkspace();
         Module module = CoronaIdeCore.getModule();
-        Datastore<WorkspaceMetaData> datastore = Datastores.getWorkspaceDatastore();
-        Path workingDirectoryName = coreConfiguration.getWorkingDirectoryName();
+        Datastore<WorkspaceMetaData> datastore = com.coronaide.core.internal.datastore.impl.Datastores
```
Comment (nickavv) [inline corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java]: pretty sure you don't need the fully qualified name here

File: corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java [hunk 2]
```diff
@@ -152,7 +146,8 @@ private void removeFromWorkspace(Workspace workspace, Project project) {
         Objects.requireNonNull(project);
 
         Module module = CoronaIdeCore.getModule();
-        Datastore<WorkspaceMetaData> datastore = Datastores.getWorkspaceDatastore();
+        Datastore<WorkspaceMetaData> datastore = com.coronaide.core.internal.datastore.impl.Datastores
```
Comment (nickavv) [inline corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java]: or here

File: corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java (hunk 1 above)
Comment (romeara) [inline corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java]: That's next - the fully-qualified name Datastores will become CoreDatastores in the API project

Comment (nickavv) [inline corona-ide-core/src/main/java/com/coronaide/core/service/impl/ProjectService.java]: alright, good answer

**Pull request merged**